### PR TITLE
cmd: add `surgery abandon-freelist` command

### PR DESCRIPTION
**Pull request by ahrtr**
Fix comment https://github.com/etcd-io/bbolt/pull/417#discussion_r1148159676

Linked to https://github.com/etcd-io/bbolt/issues/370

### CODE REVIEW

File: cmd/bbolt/command_surgery_cobra.go [hunk 1]
```diff
@@ -78,3 +78,39 @@ func surgeryClearPageElementFunc(cmd *cobra.Command, args []string) error {
 	fmt.Fprintf(os.Stdout, "All elements in [%d, %d) in page %d were cleared\n", surgeryStartElementIdx, surgeryEndElementIdx, surgeryPageId)
 	return nil
 }
+
+func newSurgeryAbandonFreelistCommand() *cobra.Command {
+	abandonFreelistCmd := &cobra.Command{
+		Use:   "abandon-freelist <bbolt-file> [options]",
+		Short: "abandon the freelist from both meta pages",
```
Comment (jmhbnz) [inline cmd/bbolt/command_surgery_cobra.go:85]: Nit - Capital "A" to start the sentence and stay consistent with the existing command.

File: cmd/bbolt/command_surgery_cobra.go [hunk 2]
```diff
@@ -78,3 +78,39 @@ func surgeryClearPageElementFunc(cmd *cobra.Command, args []string) error {
...
+
+	if err := copyFile(srcDBPath, surgeryTargetDBFilePath); err != nil {
+		return fmt.Errorf("[abandon-freelist] copy file failed: %w", err)
+	}
+
+	if err := surgeon.ClearFreelist(surgeryTargetDBFilePath); err != nil {
+		return fmt.Errorf("abandom-freelist command failed: %w", err)
+	}
+
+	fmt.Fprintf(os.Stdout, "The freelist was abandoned in boht meta pages.\nIt may cause some delay on next startup because bbolt needs to scan the whole db to reconstruct the free list.\n")
```
Comment (jmhbnz) [inline cmd/bbolt/command_surgery_cobra.go:114]: Typo here for "both"?

File: cmd/bbolt/command_surgery_cobra.go (hunk 1 above)
Comment (ahrtr) [reply cmd/bbolt/command_surgery_cobra.go:85]: updated & thx

File: cmd/bbolt/command_surgery_cobra.go (hunk 2 above)
Comment (ahrtr) [reply cmd/bbolt/command_surgery_cobra.go:114]: good catch & thx

**Comment by ahrtr** [comment]
cc @ptabor

File: cmd/bbolt/command_surgery_cobra.go [hunk 3]
```diff
@@ -78,3 +78,39 @@ func surgeryClearPageElementFunc(cmd *cobra.Command, args []string) error {
 	fmt.Fprintf(os.Stdout, "All elements in [%d, %d) in page %d were cleared\n", surgeryStartElementIdx, surgeryEndElementIdx, surgeryPageId)
 	return nil
 }
+
+func newSurgeryAbandonFreelistCommand() *cobra.Command {
+	abandonFreelistCmd := &cobra.Command{
+		Use:   "abandon-freelist <bbolt-file> [options]",
```
Comment (ptabor) [inline cmd/bbolt/command_surgery_cobra.go:84]: Maybe we prefer to be more structured: 
```bbolt surgery freelist abandon ....```
this way we could add in a future: 
```bbolt surgery freelist rebuild/check ....```

Comment (ahrtr) [reply cmd/bbolt/command_surgery_cobra.go:84]: Great point!

Comment (ahrtr) [reply cmd/bbolt/command_surgery_cobra.go]: Done. PTAL, thx

**Review by ptabor — APPROVED** [review]
Thanks.

**Comment by ahrtr** [comment]
thx @ptabor

**Pull request merged**